beams/public/js/job_requisition.js

```diff
@@ -34,5 +34,25 @@ frappe.ui.form.on('Job Requisition', {
                 }
             );
         }
-    }
+    },
+    onload: function(frm) {
+       if (!frm.doc.requested_by) {
+           // Fetch the Employee linked to the current User
+           frappe.call({
+               method: "frappe.client.get_value",
+               args: {
+                   doctype: "Employee",
+                   filters: {
+                       user_id: frappe.session.user
+                   },
+                   fieldname: "name"
+               },
```

Comment:
use get_value AJAX call. refer [this](https://frappeframework.com/docs/user/en/api/server-calls)